### ISSUE TITLE
Psalm does not warn if passed in closure has too many required arguments

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#1540]
Expected: should warn about the argument `function (int $x) : int` being incompatible, because it requires too many mandatory parameters
Observed: no warning is emitted - the other warnings are what I'd expect

```php
<?php

/**
 * @param Closure():int $x
 */
function accept_closure($x) : void {
    $x();
    $x('extra');  // properly warns
}
// Fails to warn when the passed in closure has more mandatory arguments
accept_closure(function (int $x) : int { echo "in closure;"; return 0; });
accept_closure(function () : string { return "in closure;"; });  // does warn
```

**Comment by muglug** [comment]
Thanks for this, found a bunch of new bugs
